packages/starlight-typedoc/tests/unit/typedoc.test.ts

```diff
@@ -91,3 +91,69 @@ test('should not add `README.md` module files for multiple entry points', async
   expect(writeFileSyncSpy).toHaveBeenCalled()
   expect(filePaths.some((filePath) => filePath.endsWith('README.md'))).toBe(false)
 })
+
+test('should support overriding typedoc-plugin-markdown readme and index page generation', async () => {
+  await generateTypeDoc({
+    ...starlightTypeDocOptions,
+    typeDoc: {
+      readme: 'README.md',
+      skipIndexPage: false,
+    },
+    entryPoints: ['../../fixtures/src/Bar.ts', '../../fixtures/src/Foo.ts'],
+  })
+
+  const writeFileSyncSpy = vi.mocked(fs.writeFileSync)
+  const filePaths = writeFileSyncSpy.mock.calls.map((call) => call[0].toString())
+
+  expect(filePaths.some((filePath) => filePath.endsWith('API.md'))).toBe(true)
+  expect(filePaths.some((filePath) => filePath.endsWith('README.md'))).toBe(true)
+})
+
+test('should output modules with index', async () => {
+  await generateTypeDoc({
+    ...starlightTypeDocOptions,
+    typeDoc: {
+      outputFileStrategy: 'modules',
+      entryFileName: 'index.md',
+      skipIndexPage: false,
+      flattenOutputFiles: true,
+    },
+    entryPoints: ['../../fixtures/src/module.ts'],
+  })
+
+  const writeFileSyncSpy = vi.mocked(fs.writeFileSync)
+  const filePaths = writeFileSyncSpy.mock.calls.map((call) => call[0].toString())
+
+  expect(filePaths.some((filePath) => filePath.endsWith('index.md'))).toBe(true)
+  expect(filePaths.some((filePath) => filePath.endsWith('Namespace.bar.md'))).toBe(true)
+  expect(filePaths.some((filePath) => filePath.endsWith('Namespace.foo.md'))).toBe(true)
+  expect(filePaths.some((filePath) => filePath.endsWith('Namespace.functions.md'))).toBe(true)
+  expect(filePaths.some((filePath) => filePath.endsWith('Namespace.shared.md'))).toBe(true)
+  expect(filePaths.some((filePath) => filePath.endsWith('Namespace.types.md'))).toBe(true)
+})
```

Comment:
Is this test relevant to this PR? I may be wrong but I think the changes in this PR won't change the output of this test?

Reply:
It's not strictly for anything here, it was one I just added in to check the files were generated as needed (and tbh to first get my head wrapped around vitest a little better).

But this is something that's all handled up stream and not here so can strip them out if wanted, didn't really think about it before PR-ing.